T1 login refactoring

rasakap opened the pull request.

rasakap committed:
T1 login refactoring

rasakap committed:
Merge branch 'Rasa_Kapacinskaite/develop' into Rasa_Kapacinskaite/feature/T9-replace_to_ID

rasakap committed:
Refactored code for T1 & T2

DavidV44 reviewed (changes requested):
There is no` loginPage.js` file

rasakap committed:
LoginPage.js added

DavidV44 commented on cypress/objects/LoginPage.js:
```diff
@@ -0,0 +1,63 @@
+        return cy.get(`[aria-label="${userRole}"]`)
+    }
+
+    url() {
```
`url` does not tell me what this method does, i suggest naming it `getUrl`

DavidV44 commented on cypress/objects/LoginPage.js:
```diff
@@ -0,0 +1,63 @@
+        return cy.get('.tile.form')
+    }
+
+    getUserInfoTitle(userName) {
```
Methods like these could be put in `common.js` or `pageComponents.js` files, because this is shared between pages and if you would want to check username or navigation bar it would look weird if you have to use `LoginPage `to do so.

rasakap committed:
Created a new class 'common' as after the CR;
Moved 'getInfoFormTitle' to class 'common';
Fixed the redirection of 'getInforFormTitle' from loginPage to common;

DavidV44 commented on cypress/objects/LoginPage.js:
```diff
@@ -0,0 +1,59 @@
+        return cy.get('.tile.form')
+    }
+
+        getMainNavigationBar() {
```
This also should be in common

rasakap committed:
Hot fixes of geturl object

DavidV44 commented on cypress/objects/LoginPage.js:
```diff
@@ -0,0 +1,59 @@
+        return cy.url()
+    }
+
+    getPageTitle() {
```
This also should be in common

DavidV44 commented on cypress/objects/LoginPage.js:
```diff
@@ -0,0 +1,59 @@
+        return cy.get(`[aria-label="${userRole}"]`)
+    }
+
+    geturl() {
```
This also should be in common

rasakap committed:
Moved objects from LoginPage to Common;
Changed the import place to common;

DavidV44 reviewed (approved):
pretty cool 😎
